homeassistant/components/goalzero/sensor.py

```diff
@@ -2,28 +2,137 @@
 
-from . import YetiEntity
-from .const import (
-    ATTR_DEFAULT_ENABLED,
-    DATA_KEY_API,
-    DATA_KEY_COORDINATOR,
-    DOMAIN,
-    SENSOR_DICT,
+from . import Yeti, YetiEntity
+from .const import DATA_KEY_API, DATA_KEY_COORDINATOR, DOMAIN
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="wattsIn",
+        name="Watts In",
+        device_class=DEVICE_CLASS_POWER,
+        unit_of_measurement=POWER_WATT,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="ampsIn",
+        name="Amps In",
+        device_class=DEVICE_CLASS_CURRENT,
+        unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="wattsOut",
+        name="Watts Out",
+        device_class=DEVICE_CLASS_POWER,
+        unit_of_measurement=POWER_WATT,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="ampsOut",
+        name="Amps Out",
+        device_class=DEVICE_CLASS_CURRENT,
+        unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="whOut",
+        name="WH Out",
+        device_class=DEVICE_CLASS_ENERGY,
+        unit_of_measurement=ENERGY_WATT_HOUR,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="whStored",
+        name="WH Stored",
+        device_class=DEVICE_CLASS_ENERGY,
+        unit_of_measurement=ENERGY_WATT_HOUR,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="volts",
+        name="Volts",
+        device_class=DEVICE_CLASS_VOLTAGE,
+        unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="socPercent",
+        name="State of Charge Percent",
+        device_class=DEVICE_CLASS_BATTERY,
+        unit_of_measurement=PERCENTAGE,
+    ),
+    SensorEntityDescription(
+        key="timeToEmptyFull",
+        name="Time to Empty/Full",
+        device_class=TIME_MINUTES,
+        unit_of_measurement=TIME_MINUTES,
+    ),
+    SensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        unit_of_measurement=TEMP_CELSIUS,
+    ),
+    SensorEntityDescription(
+        key="wifiStrength",
+        name="Wifi Strength",
+        device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
+        unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+    ),
+    SensorEntityDescription(
+        key="timestamp",
+        name="Total Run Time",
+        unit_of_measurement=TIME_SECONDS,
```

Comment:
A future PR should preferably calculate the absolute time when run started by taking the current time and subtracting the runtime. Then we use `DEVICE_CLASS_TIMESTAMP` here instead. We don't want relative time in the state machine since it risks spamming it with updates.

Reply:
It's not relative time (there used to be a comment in this place?). This is the time the device "ran", as in an almost brand new device 10 min now, 1 hour yesterday: 4200.

(although I wonder how useful such a sensor would be)

Reply:
Ok. 👍